sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -601,6 +601,7 @@ class SparkSession private(
   /**
    * Executes a SQL query using Spark, returning the result as a `DataFrame`.
    * The dialect that is used for SQL parsing can be configured with 'spark.sql.dialect'.
```

Comment:
I don't think we have this config anymore. Can you find out the PR that added this config, and see if there are any other places that we need to update documents?

Reply:
It seems these comment with 'spark.sql.dialect' was added in "[SPARK-13643][SQL] Implement SparkSession".
the other reference is SQLContext.scala. FYI.

Reply:
[SPARK-28997] added this config, and [SPARK-30125] removed it. For now SparkSession.sql and SQLContext.sql's comments refer to it. I think we just need to delete it?

Reply:
Yes, let's remove all the references of `spark.sql.dialect`

Reply:
done